aict_tools/io.py

```diff
@@ -263,91 +387,159 @@ def has_holes(values):
-        assert is_sorted(r.index_in_file)
-        assert not has_holes(r.index_in_file)
-        telescope_events = read_data(
-            file_path=path,
-            key=aict_config.telescope_events_key,
-            columns=telescope_event_columns,
-            first=r.index_in_file.iloc[0],
-            last=r.index_in_file.iloc[-1] + 1,
-        )
-        array_events = read_data(
-            file_path=path,
-            key=aict_config.array_events_key,
-            columns=array_event_columns,
-        )
-        df = pd.merge(left=array_events, right=telescope_events, left_on=join_keys, right_on=join_keys)
-        assert len(df) == len(telescope_events)
-        assert len(df) == len(r)
-
-    else:
+    # In case of a simple DataFrame just call read_data accordingly
+    if aict_config.data_format == 'simple':
         df = read_data(
             file_path=path,
-            key=aict_config.telescope_events_key,
+            key=aict_config.events_key,
             columns=columns,
             first=first,
             last=last,
         )
+    # For cta files multiple tables need to be read and appended/merged
+    elif aict_config.data_format == 'CTA':
+        # choose the telescope tables to load
+        file_table = tables.open_file(path)
+        if key:
+            tels_to_load = (key, )
+        else:
+            if aict_config.telescopes:
+                tels_to_load = [
+                    f"/dl1/event/telescope/parameters/{tel.name}"
+                    for tel in file_table.root.dl1.event.telescope.parameters
+                    if tel.name in aict_config.telescopes
+                ]
+            else:
+                tels_to_load = [
+                    f"/dl1/event/telescope/parameters/{tel.name}"
+                    for tel in file_table.root.dl1.event.telescope.parameters
+                ]
+
+        if "equivalent_focal_length" in columns:
+            layout = pd.read_hdf(path, '/configuration/instrument/subarray/layout')
+            optics = pd.read_hdf(path, '/configuration/instrument/telescope/optics')
+            layout = layout.merge(optics, how='outer', on="name")
+ #           layout.set_index('tel_id', inplace=True)
+
+
+        # load the telescope parameter table(s)
+        tel_dfs = []
+        for tel in tels_to_load:
+            # as not all columns are located here, we cant just use
+            # columns=columns
+            tel_df = pd.read_hdf(path, tel, start=first, stop=last)
+
+            # Pointing information has to be loaded from the monitoring tables
+            # Some magic has to be performed to merge the dfs,
+            # because only the last pointing is stored with the stage-1 ctapipe tool
+            # We also need the trigger tables as monitoring is based on time not events
+            # ToDo: Verify this for real data! MC contains only one pointing
```

Comment:
We need to interpolate the pointing to the event time given the pointing monitoring info.